components/nodemanager-service/pgdb/filtering.go

```diff
@@ -102,16 +102,17 @@ func buildWhereFilter(mergeableFilters []*common.Filter, tableAbbrev string, fil
 	}
 
 	if len(tagFilters) > 0 {
-		tagCondition, err := handleTagFilters(tagFilters)
+		tagConditions, err := handleTagFilters(tagFilters)
 		if err != nil {
-			return "", errors.Wrap(err, "buildWhereFilter error building tags")
+			return "", "", errors.Wrap(err, "buildWhereHavingFilter error building tags")
 		}
-		conditions = append(conditions, tagCondition)
+		havingFilter = fmt.Sprintf("HAVING (%s)", strings.Join(tagConditions, ") AND ("))
 	}
-
-	whereFilter = fmt.Sprintf("WHERE (%s)", strings.Join(conditions, " AND "))
-	logrus.Debugf("buildWhereFilter, whereFilter=%s", whereFilter)
-	return whereFilter, nil
+	if len(conditions) > 0 {
+		whereFilter = fmt.Sprintf("WHERE (%s)", strings.Join(conditions, " AND "))
+	}
```

Comment:
Nitpick: the result of this should be the same with or without the if condition here.

Reply:
Without the if condition, whereFilter becomes `WHERE ()` when no conditions exist. This leads to bad SQL:
```
nodemanager_service=# SELECT * FROM tags WHERE();
ERROR:  syntax error at or near ")"
LINE 1: SELECT * FROM tags WHERE();
```

Reply:
Ahh, right. Good call.